test/units/galaxy/test_token.py

```diff
@@ -96,3 +101,35 @@ def test_token_from_file_missing(b_token_file):
 @pytest.mark.parametrize('b_token_file', ['file'], indirect=True)
 def test_token_none(b_token_file):
     assert GalaxyToken(token=NoTokenSentinel).get() is None
+
+
+def test_expired_keycloak_token(monkeypatch):
+    # test long-expired token
+    response1 = u'{{"access_token": "token", "issued_at": "{0}", "expires_in": "{1}"}}'.format(
+        "2020-11-27T13:50:51.925209Z", "300"
+    )
+    # test token that will expire in a few seconds
+    response2 = u'{{"access_token": "token", "issued_at": "{0}", "expires_in": "{1}"}}'.format(
+        (datetime.now(timezone.utc) - timedelta(seconds=298)).strftime(time_format), "300"
+    )
+    mock_open = MagicMock(side_effect=[StringIO(response1), StringIO(response2)])
```

Comment:
FYI `StringIO` may leave unclosed resources/file handles in memory, which is why it's better to use `with StringIO(...) as ...:` or something like that. This sort of thing tends to become evident when executing code with warnings turned into errors. Like `python3 -Werror` or pytest's `filterwarnings = error` which is typically recommended to have set up by default.

Reply:
Fixed